Getting's started: First route (#769)
front-commerce/developers.front-commerce.com: docs/01-get-started/your-first-route.mdx

<p>{frontMatter.description}</p>

> # Routes

Reviewer: **nitpick:** there might be other ways to create routes in FC (and there is in Remix too)
```suggestion
# File-based routing
```

> Starting v3.0.0, Front-Commerce uses Remix's
> [API Routes](https://remix.run/docs/en/main/guides/api-routes) to handle all the
> routing of you application. All routes are located under `app/routes` folder.

Reviewer: **issue:** this is the v3 doc. No need to mention how it was before or that it is something new in v3.

**suggestion:** 
```suggestion
Front-Commerce uses Remix. Remix uses
[API Routes](https://remix.run/docs/en/main/guides/api-routes) to handle all the
routing of you application, and display the correct page when a user navigates to a URL. All routes are located under `app/routes` folder.
```

> API Routes will use the default export of each Route file to generate the
> route's UI. Go ahead and create a new route `hello-world.tsx`:

Reviewer: **issue:** we never mention how to access the page :)

**suggestion:** 
```suggestion
API Routes will use the default export of each Route file to generate the
route's UI. Go ahead and create a new route `hello-world.tsx`, that will display content when users browse http://localhost:4000/hello-world:
```

```tsx title="app/routes/hello-world.tsx"
export default function HelloWorld() {
  return (
    <div>
      <h1>Hello world!</h1>
    </div>
  );
}
```

# Loaders

In order to display dynamic content in your routes, you will need to fetch some
data. This is done by using Remix's Routes API method `loader`.

Here is a basic example of how a `loader` could be used:

```tsx title="app/routes/hello-world.tsx"

> export const loader = () => {

Reviewer: **issue:** we should take the habit of always adding typings and use TS.

**suggestion:** type the loader and the `useLoaderData` (with `@remix-run`'s import for now because in this example we don't yet do anything specific to FC.

>   // This method will be called prior to rendering to fetch the data for your component.

Reviewer 2: ```suggestion
  // This method will be called prior to rendering your client-side route, it's generally used to fetch the data for your component(s)
```

  return { name: "Jack Doe" };
};

export default function HelloWorld() {
  const { name } = useLoaderData();

  return (
    <div>
      <h1>Hello {name}!</h1>
    </div>
  );
}
```

> 

Reviewer: **thought:** there are many other APIs for routes in Remix. I think it'd be great to have a "kitchen sink" with additional features to act a bit as a reference for developers.

This way, they'll understand how **central** routes are in a Remix (and FC) app.

**suggestion:** add a section "And many more features".
> Routes play a fundamental part in a Remix (and Front-Commerce) application. It is the glue between the URL requested by a user and the response your application will send.
> We will continue to learn how to use them for the most common actions in the next sections, but here is a preview of everything a route could do
> **Example of the Hello world page with meta tags, fake link resources etc…** 

Prompt for https://chat-remixdocs.vercel.app/ to kickstart this section
> Create a kitchen sink for a route, with all features, so a newbie can understand every responsibilities a route has

Author: Added a "full" example, based on what we already had and what Remix docs provides as examples.

You can learn more about this API in

> [Remix's Documentation](https://remix.run/docs/en/main/guides/api-routes).

Reviewer: **issue:** there are much more thing to share here, and I think we must say that we won't cover everything because we're just using native Remix features here

**suggestion:** add a `External resources` section, with several other resources that we could enrich / reevaluate over time (e.g: the one sent by Paul, https://remix.run/docs/en/main/start/tutorial#the-contact-route-ui etc…).

There is also https://interactive-remix-routing-v2.netlify.app/ which is great!
